examples/chip-tool/commands/clusters/ComplexArgument.h

```diff
@@ -173,17 +173,39 @@ class ComplexArgumentParser
             return CHIP_ERROR_INVALID_ARGUMENT;
         }
 
-        if (strlen(value.asCString()) % 2 != 0)
+        auto str         = value.asString();
+        auto size        = str.size();
+        uint8_t * buffer = nullptr;
```

Comment:
This seems to use dynamic memory ... please document the memory lifecycle of this: why can't we use RAII (it looks like the lifetime of this value exceeds the scope of the function) and what ensure that this gets properly freed.